neuro_san/internals/chat/chat_history_message_processor.py

```diff
@@ -38,6 +53,27 @@ def get_message_history(self) -> List[Dict[str, Any]]:
         """
         return self.message_history
 
+    def process_messages(self, chat_message_dicts: List[Dict[str, Any]]):
+        """
+        Convenience method for processing lists of messages.
+        :param chat_message_dicts: The messages to process.
+        """
```

Comment:
Override the method from the superclass to allow for some post-processing of the message_history

Reply:
If we have a limit on this history length, do we need to process all messages before cutting off
the tail?

Reply:
We are actually cutting off the head.  There is more relevance w/rt current conversation towards the tail.
In regards to "process all messages": yes we process all the messages before cutting off the head.
That's the call to the superclass method just below.